exercises/collatz-conjecture/example.rs

```diff
@@ -0,0 +1,21 @@
+pub fn collatz_positive(n: u64) -> u64 {
+    if n == 1 {
+        0
+    } else  {
+        1 + match n % 2 {
+            0 => collatz_positive(n / 2),
+            _ => collatz_positive(n * 3 + 1)
+        }
+    }
+}
+
+// return Ok(x) where x is the number of steps required to reach 1
+pub fn collatz(n: u64) -> Result<u64, &'static str> {
+    if n < 1 {
+        Err("Only positive numbers are allowed")
+    } else if n == 1 {
```

Comment:
I have high confidence this `else if n == 1` is not necessary. Let me know if I am wrong on that.

Reply:
If I remove the else, then I must use "return Err(..." in the clause above.

Reply:
I agree that what you say is true, but I am still thinking lines 16 and 17 are not necessary.